secure/net/src/lib.rs

```diff
@@ -499,12 +501,14 @@ impl NetworkStream {
     /// This wraps around that buffer and blocks until all the data has been pushed.
     fn write_all(&mut self, data: &[u8]) -> Result<(), Error> {
         let mut unwritten = data;
-        let mut total_written = 0;
+        let mut total_written: u8 = 0;
```

Comment:
I think the problem you're facing is that `u8` isn't large enough. This function is used to write data on a network stream, so it's possible you're hitting `u8::MAX` and this function is failing. Change this to `u64` -- I think that should fix the error here.